### ISSUE TITLE
New block: core/transcript

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#44539]
## What problem does this address?

A text transcript is an important type of alternative content for timed media, like audio and video. WordPress offers standard blocks for displaying audio and video, but does not provide a standardized way to display transcripts. 

## What is your proposed solution?

A transcript block could ensure a stable and standardized way to add alternative content for timed media. 

For the disclosure model, the transcript block could consist of a `details` and `summary` interaction, where the `summary` element has a contained heading. The heading should be configurable by the user, but by default should either be `h2` or `h3`.

The details/summary pattern isn't fully mature, but does have broad support. I would still suggest using it rather than rolling a custom disclosure widget, thinking long term. See: https://www.scottohara.me//blog/2022/09/12/details-summary.html

E.g.

```
<details>
    <summary><h2>Transcript</h2></summary>
    <div>
       <p>Text content.</p>
     </div>
</details>
```

**Comment by carolinan** [comment]
I could see more use cases for a details summary block, perhaps Transcript could be an alias name for such a block.

This block was suggested in the past but [the issue was closed](https://github.com/WordPress/gutenberg/issues/14252) because of [limited support in IE 11](https://caniuse.com/details). I don't believe that is valid anymore. 

What inner blocks should be allowed?
